hydra-node/test/Hydra/Chain/Direct/WalletSpec.hs

```diff
@@ -227,6 +232,37 @@ isBalanced utxo originalTx balancedTx =
 ledgerPParams :: PParams (ShelleyLedgerEra Era)
 ledgerPParams = toLedgerPParams (shelleyBasedEra @Era) Fixture.pparams
 
+prop_picksCorrectUTxOToPayTheFees :: Property
+prop_picksCorrectUTxOToPayTheFees =
+  forAllBlind genUTxO $ \lookupUTxO ->
+    forAllBlind genMarkedUTxO $ \fuelUTxO ->
+      case findFuelOrLargestUTxO (Map.union lookupUTxO fuelUTxO) of
+        Nothing ->
+          property False
+            & counterexample ("Lookup UTXO: \n" <> decodeUtf8 (encodePretty lookupUTxO))
+            & counterexample ("Fuel UTXO: \n" <> decodeUtf8 (encodePretty fuelUTxO))
```

Comment:
Could: Instead of doing the `counterexample` + `forAllBlind`, you could use `forAllShow` with these nicer strings you create here.